File: .sling/libs/sushi/eqela/jk-net/0.29.0/jk.ws.lua
jk = jk or {}

jk.ws = jk.ws or {}

jk.ws.WSProtocolSource = {}

jk.ws.WSProtocol = {}
jk.ws.WSProtocol.__index = jk.ws.WSProtocol
_vm:set_metatable(jk.ws.WSProtocol, {})

function jk.ws.WSProtocol._create()
	local v = _vm:set_metatable({}, jk.ws.WSProtocol)
	return v
end

function jk.ws.WSProtocol:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.ws.WSProtocol'
	self['_isType.jk.ws.WSProtocol'] = true
	self.source = nil
end

function jk.ws.WSProtocol:_construct0()
	jk.ws.WSProtocol._init(self)
	return self
end

function jk.ws.WSProtocol:onConnectionOpened()
end

function jk.ws.WSProtocol:onConnectionClosed()
end

function jk.ws.WSProtocol:onConnectionError(error)
end

function jk.ws.WSProtocol:onStringMessage(message)
end

function jk.ws.WSProtocol:onBinaryMessage(message)
end

function jk.ws.WSProtocol:sendString(data)
	if self.source ~= nil then
		do self.source:sendString(data) end
	end
end

function jk.ws.WSProtocol:sendBuffer(data)
	if self.source ~= nil then
		do self.source:sendBuffer(data) end
	end
end

function jk.ws.WSProtocol:closeConnection()
	if self.source ~= nil then
		do self.source:closeConnection() end
	end
end

function jk.ws.WSProtocol:getSource()
	do return self.source end
end

function jk.ws.WSProtocol:setSource(v)
	self.source = v
	do return self end
end

File: .sling/libs/web5/eqela/jk-net/0.38.0/jk.ws.js
let JkWsWSProtocolSource = {};

JkWsWSProtocolSource.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkWsWSProtocolSource"] === true;
};

let JkWsWSProtocol = function() {
	this.source = null;
};

JkWsWSProtocol.prototype._t = { "JkWsWSProtocol" : true };
JkWsWSProtocol.prototype._tobj = JkWsWSProtocol;

JkWsWSProtocol.NEW = function() {
	var v = new JkWsWSProtocol;
	return v.CTOR_JkWsWSProtocol();
};

JkWsWSProtocol.prototype.CTOR_JkWsWSProtocol = function() {
	this.source = null;
	return this;
};

JkWsWSProtocol.prototype.onConnectionOpened = function() {
};

JkWsWSProtocol.prototype.onConnectionClosed = function() {
};

JkWsWSProtocol.prototype.onConnectionError = function(error) {
};

JkWsWSProtocol.prototype.onStringMessage = function(message) {
};

JkWsWSProtocol.prototype.onBinaryMessage = function(message) {
};

JkWsWSProtocol.prototype.sendString = function(data) {
	if(this.source != null) {
		this.source.sendString(data);
	}
};

JkWsWSProtocol.prototype.sendBuffer = function(data) {
	if(this.source != null) {
		this.source.sendBuffer(data);
	}
};

JkWsWSProtocol.prototype.closeConnection = function() {
	if(this.source != null) {
		this.source.closeConnection();
	}
};

JkWsWSProtocol.prototype.getSource = function() {
	return this.source;
};

JkWsWSProtocol.prototype.setSource = function(v) {
	this.source = v;
	return this;
};

JkWsWSProtocol.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkWsWSProtocol"] === true;
};
